DataReducing.php

```diff
@@ -0,0 +1,103 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Input array must have two levels
+ *      - level 1 key types: int
+ *      - level 2 key types: int
+ *
+ * @param array $arr
+ * @param string $userInput
+ * @return array
+ */
+function removeDuplicates(array $arr, string $userInput): array
+{
+    $columnKey = $arr[0][0]; //holds the working column corresponding key, used for parsing the arrays from level 2
+    foreach ($arr[0] as $key => $value) {
+        if ($userInput === $value) {
+            $columnKey = $key;
+        }
+    }
+
+    $distinctValuesArray = [];
+    foreach ($arr as $key => $value) {
+        $distinctValuesArray[] = $arr[$key][$columnKey];
+    }
+    $distinctValuesArray = array_unique($distinctValuesArray);
+
+    $resultArray = [];
+    foreach ($distinctValuesArray as $key => $columnValue) {
+        $foundFirstAppearance = false;
+        foreach ($arr as $entryKey => $v) {
+            if ($arr[$entryKey][$columnKey] === $columnValue) {
+                $resultArray[]        = $arr[$entryKey];
+                $foundFirstAppearance = true;
+            }
+            if ($foundFirstAppearance) {
+                break;
+            }
+        }
+    }
+
+    return $resultArray;
+}
+
+/**
+ * @param array $arr
+ * @param string $userInput
+ * @return array
+ */
+function filterByWhere(array $arr, string $userInput): array
+{
+    $wordPattern = '/([\w]+)/u';
```

Comment:
how about, after you know the where symbol, explode the input and you get the column and the value in the result.

Reply:
solved